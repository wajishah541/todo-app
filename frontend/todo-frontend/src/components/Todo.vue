<template>
  <form v-on:submit.prevent>
    <h1>{{ title }}</h1>
    <label>Add todo here </label>
    <input type="text" required v-model="task" />
    <button class="add" @click="add">Add</button>
    <div v-for="(task, index) in tasks" :key="index">
      <ul>
        <li>
          {{ task.content }}
        </li>
      </ul>
    </div>
  </form>
</template>
<script>
export default {
  name: "Todo",
  props: {
    title: {
      type: String,
      required: true,
    },
  },
  data() {
    return {
      task: "",
      tasks: [
        {
          content: task,
          ID: true,
        },
        {
          content: task,
          ID: true,
        },
      ],
    };
  },
  methods: {
    add() {
      this.tasks.push({
        content: this.task,
      });
      this.task = "";
    },
  },
};
</script>
<style>
h1 {
  text-align: center;
}
form {
  max-width: 420px;
  margin: 30px auto;
  background: white;
  text-align: left;
  padding: 40px;
  border-radius: 40px;
}
label {
  color: black;
  display: inline-block;
  margin: 25px 0 15px;
  font-size: 0.6em;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: bold;
}
table {
  width: 100%;
  margin: 30px auto;
  background: white;
  text-align: left;
  padding: 10px;
  border-radius: 5px;
}
input {
  color: black;
  display: block;
  padding: 10px 6px;
  width: 100%;
  box-sizing: border-box;
  border: none;
  border-bottom: 1px solid #ddd;
}

button {
  background-color: #4caf50;
  border: none;
  color: black;
  padding: 10px 22px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  width: 90%;
  margin: 10px;
  font-weight: bold;
}
</style>
