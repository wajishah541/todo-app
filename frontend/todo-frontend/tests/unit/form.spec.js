import { shallowMount } from '@vue/test-utils'
import Todo from '../../src/components/Todo.vue'
import axios from 'axios'
import sinon from 'sinon'
// Acceptance Test
describe('The todo.vue component', () => {
  it('Enables user to add todo ', () => {
    const wrapper = shallowMount(Todo, {
      props: {
        title: 'Todo-App'
      }
    })
    const tasks = [
      { ID: 1, content: 'go for a walk ' }
    ]
    axios.post = jest.fn().mockResolvedValue({
      data: [
        {
          ID: 1, content: 'go for a walk '
        }
      ]
    })
    const result = insertTodo()
    expect(axios.post).toHaveBeenCalledWith(`${postTodoUrl}/handlerequest`)
    expect(result.data).toEqual(tasks)
  })
  // component test
  describe('The todo.vue component', () => {
    // component shallow rendering test
    it('renders a component', () => {
      const wrapper = shallowMount(Todo, {
        propsData: {
          title: 'Todo-App'
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
    // Test case for title
    it('Displays the title when passed as a prop', () => {
      const wrapper = shallowMount(Todo, {
        propsData: {
          title: 'My TodoApp'
        }
      })
      // expect(wrapper.text()).toMatch('My TodoApp')
      expect(wrapper.props().title).toBe('My TodoApp')
      expect(wrapper.props('title')).toBe('My TodoApp')
    })
    // test case for todo input field
    it('allows for adding  todo in todo-input field ', async () => {
      const wrapper = shallowMount(Todo, {
        propsData: {
          title: 'Todo-App'
        }
      })
      const input = wrapper.find('input')
      await input.setValue('buy some milk')
      expect(wrapper.find('input[type="text"]').element.value).toBe('buy some milk')
    })
    // test case for button
    it('triggers on click', async () => {
      const wrapper = shallowMount(Todo, {
        propsData: {
          title: 'Todo-App'
        }
      })
      await wrapper.find('button').trigger('click')
    })
  })
  it('click on add button calls our method', async () => {
    const wrapper = shallowMount(Todo, {
      propsData: {
        title: ''
      },
    })
    const spy = jest.spyOn(wrapper.vm, 'add');
    wrapper.vm.add();
    expect(spy).toHaveBeenCalled();
  })
})
