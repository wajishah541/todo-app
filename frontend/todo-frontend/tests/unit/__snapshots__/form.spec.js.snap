// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The todo.vue component The todo.vue component renders a component 1`] = `
"<form>
  <h1>Todo-App</h1><label>Add todo here </label><input type=\\"text\\"><button>Add</button>
</form>"
`;
